### DISCUSSION TITLE
Why GetReactingAtoms returns inaccurate changing atoms?

### DISCUSSION THREAD

**Discussion by yangyinuo823** [rdkit/rdkit#4552] [Q&A]
Hi everyone,

I really need help with function GetReactingAtoms. In the below reaction, the changing atoms should  be [n:5] [n:6] [H:9] atoms, which should give some output like (4, 5, 8). However, what I get is ((5,6,7),), which makes no sense. This mismatch happens in a lot of my reactions, this is only one example. Did I do anything wrong? Thanks in advance for any sugguestions!!

Input:
```
from rdkit import Chem
from rdkit.Chem import rdChemReactions
rxn_str = '[O:1]([c+:2]1[n-:3][c:4]([H:8])[n:5][n:6]1[H:9])[H:7]>>[O:1]([c:2]1[n:3][c:4]([H:8])[n:5]([H:9])[n:6]1)[H:7]'
rxn = rdChemReactions.ReactionFromSmarts(rxn_str,useSmiles=True)
rxn.Initialize()
print(rxn.GetReactingAtoms(mappedAtomsOnly=True))
```

Output:
```
((5, 6, 7),)
```

**Comment by greglandrum** [comment]
Hi @yangyinuo823, can you explain why you expect the output to be `(4,5,8)`?

**Reply by yangyinuo823** [reply]
First thank you for reply! 
Because the chaning atoms are the 5th, 6th, 9th atoms and since the result counts starting from zero, so I thought the output should be (4,5,8).

**Reply by yangyinuo823** [reply]
To be more clear, this reaction breaks N5-H9 bond and forms N6-H9 bond, that's why the changing atoms are the 5th, 6th, 9th atoms.

**Reply by yangyinuo823** [reply]
I'm sorry for my stupid question! After I extract the fragment of reacting atoms, it turns out of be correct!
The reacting atoms index are (5,6,7), and the corresponding fragment is [n:5][n:6][H:9].
Does this mean the atom index can be different from the atom mapping index?

**Reply by greglandrum** [reply]
Exactly. It’s the indices of the atoms in the reactant template, not the atom map numbers

**Reply by yangyinuo823** [reply]
That answers my confusion, thank you!
